tests/rules/taint_exception.py

```diff
@@ -0,0 +1,242 @@
+# This test aims to exercise various exception paths.
+
+def throw_exits(input):
+  clean = None
+  raise RuntimeError()
+
+  # ok:
+  sink = clean
+
+def throw_in_catch_exits(input):
+  clean = None
+  try:
+    raise RuntimeError()
+  except Exception as e:
+    raise RuntimeError()
+
+  #ok: unreachable, because throw inside the catch clause exits
+  sink(clean)
+
+def throw_in_else_exits(input):
+  clean = None
+  try:
+    pass
+  except Exception as e:
+    clean = input
+    #ok: unreachable, because exception was not thrown
```

Comment:
I wonder if we're trying to be too smart here. Who writes this code? And if they write it, is it really that bad that we report a finding here?

Reply:
Could you point me where in the code we make this unreachable?

Reply:
Ah OK I see, now the arcs to the `catch` are only added via `throw_destination`. Fine with that, I think it deserves a comment where we used to connect `finaltry` with `catchesi`, like "we do not directly connect finaltry with catchesi but whenever we encounter a Call or a Throw inside a Try, we add an arc ...".

Reply:
Will add a comment to explain this.